pkg/humio/client.go

```diff
@@ -106,37 +107,53 @@ type LicenseClient interface {
 
 // ClientConfig stores our Humio api client
 type ClientConfig struct {
-	apiClient *humioapi.Client
-	logger    logr.Logger
-	userAgent string
+	apiClient    *humioapi.Client
+	logger       logr.Logger
+	userAgent    string
+	humioClients map[humioClientKey]*humioapi.Client
+}
+
+type humioClientKey struct {
+	namespace, name string
+	authenticated   bool
 }
 
 // NewClient returns a ClientConfig
 func NewClient(logger logr.Logger, config *humioapi.Config, userAgent string) *ClientConfig {
 	client := humioapi.NewClient(*config)
 	return &ClientConfig{
-		apiClient: client,
-		logger:    logger,
-		userAgent: userAgent,
+		apiClient:    client,
+		logger:       logger,
+		userAgent:    userAgent,
+		humioClients: map[humioClientKey]*humioapi.Client{},
 	}
 }
 
 // SetHumioClientConfig takes a Humio API config as input and ensures to create a new API client that uses this config
-func (h *ClientConfig) SetHumioClientConfig(config *humioapi.Config, overrideExistingConfig bool) {
-	if !overrideExistingConfig {
-		if config.Token == "" {
-			config.Token = h.apiClient.Token()
-		}
-		if config.Address == nil {
-			config.Address = h.apiClient.Address()
-		}
-		if config.CACertificatePEM == "" {
-			config.CACertificatePEM = h.apiClient.CACertificate()
+func (h *ClientConfig) SetHumioClientConfig(config *humioapi.Config, req ctrl.Request) {
+	config.UserAgent = h.userAgent
+	key := humioClientKey{
+		namespace:     req.Namespace,
+		name:          req.Name,
+		authenticated: config.Token != "",
+	}
+	c := h.humioClients[key]
+	if c == nil {
+		c = humioapi.NewClient(*config)
+	} else {
+		existingConfig := c.Config()
+		equal := existingConfig.Token == config.Token &&
+			existingConfig.Insecure == config.Insecure &&
+			existingConfig.CACertificatePEM == config.CACertificatePEM &&
+			existingConfig.ProxyOrganization == config.ProxyOrganization &&
+			existingConfig.Address.String() == config.Address.String()
+		if !equal {
+			//h.logger.Info("recreating Humio API client")
```

Comment:
Do you mean to have this commented out?

Reply:
Ah, no. I commented that out with the intention to delete it later, but looks like I forgot to do that. I've force pushed a commit that gets rid of this line. Thanks!